core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -610,6 +611,13 @@ public boolean isReady() {
       }
       return realStream.isReady();
     }
+
+    @Override
+    public void setMessageCompression(boolean enable) {
+      if (realStream != null) {
+        realStream.setMessageCompression(enable);
```

Comment:
This doesn't seem to be enough. We'll have to record what the compression was for each message or some such.